File: Anul II/Medii de proiectare si programare/Laborator/lab05/Ubb.BikeContest.Service/src/ParticipantService.cs
using System.Collections;
using Ubb.BikeContest.Model;
using Ubb.BikeContest.Repository;
using Ubb.BikeContest.Service.validation;

namespace Ubb.BikeContest.Service;

public class ParticipantService : IParticipantService
{
    private readonly IParticipantRepository _participantRepository;
    private readonly IValidator<long, Participant> _validator;

    public ParticipantService(IParticipantRepository participantRepository)
    {
        _participantRepository = participantRepository;
        _validator = new ParticipantValidator();
    }
    
    public Participant FindById(long id)
    {
        return _participantRepository.FindById(id);
    }

    public IEnumerable<Participant> FindAll()
    {
        return _participantRepository.FindAll();
    }

    public void Save(Participant newEntity)
    {
        _validator.ValidateEntity(newEntity);
        _participantRepository.Save(newEntity);
    }

    public void Delete(long id)
    {
        _participantRepository.Delete(id);
    }

    public void Update(Participant updatedEntity)
    {
        _validator.ValidateEntity(updatedEntity);
        _participantRepository.Update(updatedEntity);
    }

    public IEnumerable<Participant> GetParticipantsByTeam(long teamId)
    {
        return _participantRepository.GetParticipantsByTeam(teamId);
    }

    public Participant GetParticipantByData(Participant participant)
    {
        return _participantRepository.GetParticipantByData(participant);
    }
}

File: Anul II/Medii de proiectare si programare/Laborator/lab11/java/services/src/main/java/com/ubb/service/ParticipantService.java
package com.ubb.service;

import com.ubb.model.Participant;
import com.ubb.repository.IParticipantRepository;
import com.ubb.validation.IValidator;
import com.ubb.validation.ParticipantValidator;

import java.util.Collection;
import java.util.Optional;

public class ParticipantService implements IParticipantService {
    private final IParticipantRepository participantRepository;
    private final IValidator<Participant> validator;

    public ParticipantService(final IParticipantRepository participantRepository) {
        this.participantRepository = participantRepository;
        validator = new ParticipantValidator();
    }

    @Override
    public Collection<Participant> getParticipantsByTeam(final Long teamID) {
        return participantRepository.getParticipantsByTeam(teamID);
    }

    @Override
    public Optional<Participant> getParticipantByData(Participant participant) {
        return participantRepository.getParticipantByData(participant);
    }

    @Override
    public Optional<Participant> findById(final Long id) {
        return participantRepository.findById(id);
    }

    @Override
    public Collection<Participant> findAll() {
        return participantRepository.findAll();
    }

    @Override
    public void save(final Participant newEntity) {
        validator.validateEntity(newEntity);
        participantRepository.save(newEntity);
    }

    @Override
    public void delete(final Long id) {
        participantRepository.delete(id);
    }

    @Override
    public void update(final Participant updatedEntity) {
        validator.validateEntity(updatedEntity);
        participantRepository.update(updatedEntity);
    }
}
